### PR TITLE
Added empty help test (bsc#972548)

### PR DESCRIPTION
**Pull request by lslezak**
## Improved Tests

- The `tests/integration/run.rb` script used fixed temporary file names in `/tmp`, that's a security problem
  - Switched to the `Tempfile` objects
  - The temporary files are automatically created, so no need to check the existence
- Updated the libyui tests to dump the screenshots to the `log/` subdirectory (to not mess up the sources)
- Added unit test for https://bugzilla.suse.com/show_bug.cgi?id=972548

### CODE REVIEW

File: tests/libyui/table_sort_spec.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 require_relative "rspec_tmux_tui"
 
+log_dir = "#{__dir__}/log"
+Dir.mkdir log_dir if !File.exist?(log_dir)
```
Comment (jreidinger) [inline tests/libyui/table_sort_spec.rb:4]: I see that multiple times, can it be shared somewhere like in `rspec_tmux_tui`

**Review by jreidinger — APPROVED** [review]
in general looks good, just one NP

File: tests/libyui/empty_help_spec.rb [hunk 2]
```diff
@@ -14,23 +14,26 @@ def check_dialog
...
+  # returned :help in the ncurses UI if the help text was empty,
+  # the Qt UI also displays an empty popup
   it "displays empty popup for empty help text" do
     @tui.await("[Help]")
     check_dialog
     @tui.capture_pane_to("#{@log_base}-1-initial")
 
     # show help by pressing F1
     @tui.send_keys "F1"
+    # check that the help popup is displayed, there is an empty popup
```
Comment (shundhammer) [inline tests/libyui/empty_help_spec.rb:35]: :+1:

**Review by shundhammer — APPROVED** [review]
LGTM

**Review by jreidinger — APPROVED** [review]

**Pull request merged**